src/providers/php.rs

```diff
@@ -0,0 +1,203 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+use crate::nixpacks::{
+    environment::EnvironmentVariables,
+    nix::pkg::Pkg,
+    phase::{BuildPhase, InstallPhase, SetupPhase, StartPhase}, app::App,
+};
+
+use super::{node::NodeProvider, Provider};
+
+pub struct PhpProvider;
+
+impl Provider for PhpProvider {
+    fn name(&self) -> &str {
+        "php"
+    }
+
+    fn detect(
+        &self,
+        app: &crate::nixpacks::app::App,
+        _env: &crate::nixpacks::environment::Environment,
+    ) -> anyhow::Result<bool> {
+        Ok(app.includes_file("composer.json") || app.includes_file("index.php"))
+    }
+
+    fn setup(
+        &self,
+        app: &crate::nixpacks::app::App,
+        env: &crate::nixpacks::environment::Environment,
+    ) -> anyhow::Result<Option<crate::nixpacks::phase::SetupPhase>> {
+        let nodejs = NodeProvider {};
+
+        let php_pkg = match self.get_php_package(&app) {
+            Ok(Some(php_package)) => php_package,
+            _ => "php".to_string(),
+        };
+        let mut pkgs = vec![
+            Pkg::new(&php_pkg),
+            Pkg::new("perl"),
+            Pkg::new(&format!("{}Packages.composer", &php_pkg))
+        ];
+        if let Ok(php_extensions) = self.get_php_extensions(&app) {
+            for extension in php_extensions {
+                pkgs.push(Pkg::new(&format!("{}Extensions.{}", &php_pkg, extension)));
+            }
+        }
+
+        if let Ok(true) = nodejs.detect(app, env) {
+            if let Ok(Some(mut node_setup)) = nodejs.setup(app, env) {
+                pkgs.append(&mut node_setup.pkgs);
+            }
+        }
+
+        Ok(Some(SetupPhase::new(pkgs)))
+    }
+
+    fn install(
+        &self,
+        app: &crate::nixpacks::app::App,
+        env: &crate::nixpacks::environment::Environment,
+    ) -> anyhow::Result<Option<crate::nixpacks::phase::InstallPhase>> {
+        let mut cmd = String::new();
+        cmd.push_str("sudo apt-get update && sudo apt-get install -y nginx");
```

Comment:
Nginx is available on Nix and we should install it from there instead. https://search.nixos.org/packages?channel=unstable&show=nginx&from=0&size=50&sort=relevance&type=packages&query=nginx

Reply:
See #183 &mdash; in order to serve PHP, the Nginx configuration needs access to the system FastCGI configuration, which is installed with Nginx. In order to know where that is, I need to access it from within the Nix configuration.

Reply:
I see. Can you provide an example `environment.nix` file that would work so we can start getting that capability integrated into the Nix pkg struct.

Reply:
I figured out how to install Nginx through Nix without having access to `environment.nix`, with the `nix-store` command.